presto-hive/src/main/java/com/facebook/presto/hive/HiveZeroRowFileCreator.java

```diff
@@ -144,10 +149,14 @@ public void createFiles(ConnectorSession session, HdfsContext hdfsContext, Path
         }
     }
 
-    private void createFile(HdfsContext hdfsContext, Path path, byte[] content)
+    private void createFile(HdfsContext hdfsContext, Path path, byte[] content, ConnectorSession session)
     {
-        try (OutputStream outputStream = hdfsEnvironment.getFileSystem(hdfsContext, path).create(path, false)) {
-            outputStream.write(content);
+        try {
+            FileSystem fs = hdfsEnvironment.getFileSystem(hdfsContext, path);
+            DataSink dataSink = dataSinkFactory.createDataSink(session, fs, path);
+            DataOutput dataOutput = createDataOutput(Slices.wrappedBuffer(content));
+            dataSink.write(ImmutableList.of(dataOutput));
+            dataSink.close();
```

Comment:
I know that might be slightly out of scope of this PR, but I would recommend changing the `DataSink` interface to extend `Closeable` and use automatic resource management (`try (DataSink dataSink = dataSinkFactory.createDataSink(session, fs, path)) {`) to make sure it is always closed, even in case of a failure. Also maybe also fix it here if you are at it: https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/pagefile/PageFileWriterFactory.java#L124